wordpress-mobile/WordPress-iOS
# New AddSiteAlertFactory convenience class.

This PR implements a new `AddSiteAlertFactory` class to modularize some of the code that was previously located in `BlogListViewController.swift`.

The reason for this change is because of some upcoming changes, that will require this action sheet to be shown from within another VC (`BlogDetailsViewController`).

This is part of [a larger upcoming change](https://github.com/wordpress-mobile/WordPress-iOS/pull/15130), and was broken up to make the review and testing easier.

## To test:

Please remember to test this both in iPhone and iPad to make sure I haven't broken the UI.

### Test 1:

1. Log into your WP.com account
2. Go to My Sites.
3. Tap “Switch Sites” in the top left corner.
4. Tap “+”

An action sheet with 2 buttons should come up, looking like this.

![actionSheet](https://user-images.githubusercontent.com/1836005/96519636-d294ad80-1243-11eb-9961-ca1e1fff9769.png)

5. Tap “Create WordPress.com Site”

Make sure the WP.com Site creation Wizard is launched.

<img width="350" alt="Screen Shot 2020-10-19 at 19 48 40" src="https://user-images.githubusercontent.com/1836005/96519821-35864480-1244-11eb-899e-92b6217edbc1.png">

You can tap “Cancel” and proceed with the second test.


### Test 2:

1. Log into your WP.com account
2. Go to My Sites.
3. Tap “Switch Sites” in the top left corner.
4. Tap “+”
5. Tap “Add self-hosted site”

Make sure the Self-Hosted Site Address VC is shown.  It looks like this:

<img width="350" alt="Screen Shot 2020-10-19 at 19 50 08" src="https://user-images.githubusercontent.com/1836005/96519881-53ec4000-1244-11eb-9b6c-8c0bf8188d85.png">

Feel free to “Cancel”


## PR submission checklist:
